### PR TITLE
[ArrayManager] Add Series._as_manager (consolidate with DataFrame)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Move the `_as_manager` helper to `generic.py`, and ensure it works for Series as well

### CODE REVIEW

File: pandas/core/internals/construction.py [hunk 1]
```diff
@@ -202,15 +206,21 @@ def mgr_to_mgr(mgr, typ: str):
         if isinstance(mgr, BlockManager):
             new_mgr = mgr
         else:
-            new_mgr = arrays_to_mgr(
-                mgr.arrays, mgr.axes[0], mgr.axes[1], mgr.axes[0], typ="block"
-            )
+            if mgr.ndim == 2:
```
Comment (jreback) [inline pandas/core/internals/construction.py:209]: can you make elif/else

File: pandas/core/internals/construction.py [hunk 2]
```diff
@@ -202,15 +206,21 @@ def mgr_to_mgr(mgr, typ: str):
...
+                )
+            else:
+                new_mgr = SingleBlockManager.from_array(mgr.arrays[0], mgr.index)
     elif typ == "array":
         if isinstance(mgr, ArrayManager):
             new_mgr = mgr
         else:
-            arrays = [mgr.iget_values(i).copy() for i in range(len(mgr.axes[0]))]
-            new_mgr = ArrayManager(arrays, [mgr.axes[1], mgr.axes[0]])
+            if mgr.ndim == 2:
```
Comment (jreback) [inline pandas/core/internals/construction.py:219]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/construction.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/core/internals/construction.py:209]: I personally wouldn't find that an improvement in readability. Now it's very clearly separated into two cases (manager already has the same type -> use as is, or else the manager needs to be converted)

Comment (jreback) [reply pandas/core/internals/construction.py:209]: using nested elif/if is an anti-pattern

Comment (jorisvandenbossche) [reply pandas/core/internals/construction.py:219]: It's only an anti-pattern if it's deeply nested and hampering readability. In this case, it makes the code easier to understand IMO.

**Review by jreback — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks

**Pull request merged**